### DISCUSSION TITLE
Question: Is it possible to specify the size of the k-buckets?

### DISCUSSION THREAD

**Discussion by cndolo** [anacrolix/dht#39] [Q&A]
The title really states it all.
My current use case: I want to run a DHT node that remembers as many good nodes as possible.
Allowing the k-buckets to be (infinitely) large results in good nodes not being forgotten once they are stored in the routing table.

**Comment by anacrolix** [comment]
The machinery is all there to support it, but the configuration is not exposed. I would accept a PR if you're interested in doing it. Curiously, early versions of this module didn't have a limit to the node table size, or bucketize at all (it just kept everything and grew unbounded).

You might tap in to one of the callbacks for when a response is received, this indicates a peer is good. Then you are free to do what you want with it, including maintaining your own table separate to the server and managing it with calls in to the server.
